### PR TITLE
Js console as Eval exec

### PR DESCRIPTION
**Pull request by gaurave**
* Adds a js-console as eval exec to wash
* Prints a `eval` prompt when run `eval` command
* Read input line by line and prints the result.
* spits the `error message` returned by `javascript eval` command if it fails.
* Partial multi-line input support inferring from eval error.

@rpaquay  PTAL for review
cc @rginda

### CODE REVIEW

**Comment by ussuri** [comment]
Very cool

File: lib/wash/exe/eval.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+main.signature = {
+  'input-history': '@',
+  'prompt-string|p': '$'
+};
+
+/**
+ * @return {void}
+ */
+Eval.prototype.setPrompt = function() {
+  this.promptString_ = this.tc_.output('%set-attr(FG_BOLD, FG_CYAN)' +
```
Comment (ussuri) [inline lib/wash/exe/eval.js]: Looks like this could be just one string?

File: lib/wash/exe/eval.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
+// Copyright 2014 Google Inc. All rights reserved.
```
Comment (rpaquay) [inline lib/wash/exe/eval.js]: 2015

File: lib/wash/exe/eval.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          return Promise.resolve(value);
+        }.bind(this));
+    }.bind(this)
+  ).catch(
+    function(error) {
+      // We currently do not support multi line input in readline. This is a
+      // proxy to find if the error is a syntax error in code or the user has
+      // not finished typing.
+      // TODO(grv): Do a better detection of syntax error vs incomplete code.
+      if (error && error.message == 'Unexpected end of input') {
```
Comment (rpaquay) [inline lib/wash/exe/eval.js]: This seems fragile. Where is this message coming from? Can't we wrap it at the call site and throw a custom exception?

At the very least, I would add a note explaining where the message is coming from.

File: lib/wash/exe/eval.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ * @param {string} input
+ * @return {Promise<*>}
+ */
+Eval.prototype.evaluate = function(input) {
+  console.log(input);
+  try {
+    var value = eval(input);
+    return Promise.resolve(value);
+  } catch (err) {
+    console.log(err);
```
Comment (rpaquay) [inline lib/wash/exe/eval.js]: Is this needed? It seems the caller catches errors and print them...

File: lib/wash/exe/eval.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+          });
+    }.bind(this));
+};
+
+/**
+ * @param {string} input
+ * @return {Promise<*>}
+ */
+Eval.prototype.evaluate = function(input) {
+  console.log(input);
```
Comment (rpaquay) [inline lib/wash/exe/eval.js]: Remove?

File: lib/wash/exe/eval.js (hunk 4 above)
Comment (rpaquay) [inline lib/wash/exe/eval.js]: Nevermind, it is a console.log (I thought it was a stdout.write). Remove it then?

File: lib/wash/exe/eval.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+  this.exit = false;
+
+  this.promptString_ = executeContext.getArg('prompt-string', '');
+  this.promptVars_ = null;
+
+  this.inputHistory = [];
+};
+
+export {Eval};
```
Comment (rpaquay) [inline lib/wash/exe/eval.js]: Is this needed?

**Pull request merged**